### PR TITLE
Load Product Page V2 Feature flag at install

### LINKED ISSUE #23999: [Add support for experimental features] Handle Product Page V2 flag setup
This issue is an item in the [Add support for experimental features](https://github.com/PrestaShop/PrestaShop/issues/23235) project

You can read about the project [here](https://github.com/PrestaShop/PrestaShop/issues/23235).

### Handle Product Page V2 flag setup at install

AS a merchant
WHEN I Install PrestaShop 1.7.8
AND I open the Back Office 'Experimental Features' BO page after the install process was successfull
THEN I see only one Feature flag: the one used for 'Experimental Product Page'

### Handle Product Page V2 flag setup at upgrade

AS a merchant
WHEN I upgrade from PrestaShop < 1.7.8 to PrestaShop 1.7.8
AND I open the Back Office 'Experimental Features' BO page after the install process was successfull
THEN I see only one Feature flag: the one used for 'Experimental Product Page'

## Needed

- the _exact_ wording for Product Page V2 flag: content and translation domain
- the _exact_ help text for Product Page V2 flag: content and translation domain

tdg

The exact wording is item A.

The exact help text is item B.

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Load Product Page V2 Feature flag at install using `db_data.sql` feature (see https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Install/Install.php#L508). Load Product Page V2 Feature flag at upgrade using standard 1.7.8 SQL script. Also fixes a wording domain.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/23999
| How to test?      | See below
| Possible impacts? | I changed the 1.7.7 -> 1.7.8 SQL upgrade script, the database install script and one page title.

## How to test

1. Install PS using this branch. Verify that when you visit BO page Configure > Advanced Parameters
Experimental Features you can see the Product Page V2 feature flag, that is named `Experimental product page`

2. Install PS 1.7.7 then perform an upgrade to 1.7.8 using this branch. Verify that when you visit BO page Configure > Advanced Parameters
Experimental Features you can see the Product Page V2 feature flag, that is named `Experimental product page`

## Screenshot

Capture d’écran 2021-04-13 à 22 13 51

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24037)

### CODE REVIEW

**Comment by matks** [comment]
CI failed with error message `Error: API rate limit exceeded for installation ID 572029.`

**Comment by kpodemski** [comment]
Wouldn't be better to use `ps_translation` for such a translatable elements? 🤔 I guess this should be some kind of example for future new controllers like this one? PrestaShop has standarized translations for a very long time where every time you want to have translatable entity you have relation between `entity_lang` table, this flat table seems a bit odd to me

Edit: never mind, I guess this scenario is specific as we have predefined texts, not texts available to edit by user

**Review by Progi1984 — DISMISSED** [review]

**Comment by matks** [comment]
> Edit: never mind, I guess this scenario is specific as we have predefined texts, not texts available to edit by user

Exactly 😄  this was done on purpose so that the translation wordings can be managed on Crowdin and when user installs a localization pack, it has the names "Experimental Features" and so already translated.

**Review by kpodemski — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @matks,

```php
1. Install PS using this branch. Verify that when you visit BO page Configure > Advanced Parameters
Experimental Features you can see the Product Page V2 feature flag, that is named Experimental product page
```
=> ok :heavy_check_mark: 
image

```php
2. Install PS 1.7.7 then perform an upgrade to 1.7.8 using this branch. Verify that when you visit BO page Configure > Advanced Parameters
Experimental Features you can see the Product Page V2 feature flag, that is named Experimental product page
```
I made an upgrade from Fresh PS1773 to PS178(using PR) in PHP7.1 using the 1-click upgrade v4.12.0
=>
1. NOK, I have errors
image
Here's my log
[log.txt](https://github.com/PrestaShop/PrestaShop/files/6310299/log.txt)
2. After the upgrade, I tried to access to BO > Advanced Parameters > Experimental Features page => No feature flag is displayed 
image

Thank you!

**Review by Progi1984 — APPROVED** [review]

**Comment by matks** [comment]
Hi @khouloudbelguith I have fixed the upgrade issue thanks to @atomiix 

I did small database upgrade and full upgrade process it seems to be OK

Capture d’écran 2021-04-14 à 14 54 17
Capture d’écran 2021-04-14 à 15 03 16

File: install-dev/data/db_data.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+SET
+  SESSION sql_mode='';
+SET
+  NAMES 'utf8mb4';
+
+INSERT INTO `PREFIX_feature_flag` (`name`, `state`, `label_wording`, `label_domain`, `description_wording`, `description_domain`)
+VALUES
+	('product_page_v2', 0, 'Experimental product page', 'Admin.Advparameters.Feature', 'This page is a work in progress. It includes new combination management features and other features under development (virtual products, packs, etc.)', 'Admin.Advparameters.Help');
```
Comment (atomiix) [inline install-dev/data/db_data.sql:8]: Why not using xml files here?

Comment (matks) [reply install-dev/data/db_data.sql:8]: XML files are demo fixtures, which means the merchant can choose NOT to install them.

We dont want merchant to choose whether or not this data is inserted in database 😄  . We want to be sure it is inserted all the time.

Also even if we decided to use XML files to insert this flag, it is likely we will remove the flag at 1.7.8->1.7.9 step when Product Page V2 becomes complete. So the insert would be an XML files while the removal would be SQL.

Comment (matks) [reply install-dev/data/db_data.sql:8]: Also using this db_data.sql is closer to the SQL upgrade scripts so the "insert Product Page V2 feature flag" logic is similar for install process and upgrade process. We want the flag to be inserted whether merchant installs or upgrades, so if we can keep the logic similar it eases future maintainance.

Comment (atomiix) [reply install-dev/data/db_data.sql:8]: XML for demo fixtures are in `install/fixtures/fashion/data/`, other data like user roles & rights, hooks, etc... are in `install/data/xml/`. I think it makes sense to use this way and to use a migration script when we'll want to remove it.

Comment (matks) [reply install-dev/data/db_data.sql:8]: I let you discuss this with @eternoendless he's the one who did not want XML files 😄

**Review by atomiix — APPROVED** [review]

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 2]
```diff
@@ -70,3 +70,23 @@ UPDATE `PREFIX_product` SET `product_type` = "pack" WHERE `cache_is_pack` = 1;
...
+  `label_domain` VARCHAR(255) COLLATE utf8mb4_general_ci NOT NULL DEFAULT '',
+  `description_wording` VARCHAR(191) COLLATE utf8mb4_general_ci NOT NULL DEFAULT '',
+  `description_domain` VARCHAR(255) COLLATE utf8mb4_general_ci NOT NULL DEFAULT '',
+  PRIMARY KEY (`id_feature_flag`),
+  UNIQUE KEY `UNIQ_91700F175E237E06` (`name`)
+) ENGINE=ENGINE_TYPE DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+INSERT INTO `PREFIX_feature_flag` (`name`, `state`, `label_wording`, `label_domain`, `description_wording`, `description_domain`)
+VALUES
+	('product_page_v2', 0, 'Experimental product page', 'Admin.Advparameters.Feature', 'This page is a work in progress. It includes new combination management features and other features under development (virtual products, packs, etc.)', 'Admin.Advparameters.Help');
```
Comment (eternoendless) [inline install-dev/upgrade/sql/1.7.8.0.sql:91]: This part should be removed right? It's already done in `db_data.sql`
```suggestion
```

File: install-dev/data/db_data.sql (hunk 1 above)
Comment (eternoendless) [reply install-dev/data/db_data.sql:8]: For the record: fixtures use ObjectModels, and FeatureFlag is a Doctrine entity, so you can't use them.

File: install-dev/upgrade/sql/1.7.8.0.sql (hunk 2 above)
Comment (eternoendless) [reply install-dev/upgrade/sql/1.7.8.0.sql:91]: Whoops, my mistake, I thought we were in the db_structure file

**Review by eternoendless — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @matks,

1. Fresh install PS178 => ok
https://drive.google.com/file/d/1LnG8gxUDMy0Y27nTlbr-n8i03MLuuabs/view?usp=sharing
2. Upgrade from PS1773 to PS1780 => ok
https://drive.google.com/file/d/1vjI42k_v3MXWYI0lEmZStvffQGKBCsdP/view?usp=sharing
3. Upgrade from PS1770 to PS1780 => ok
https://drive.google.com/file/d/1aUvttAP9IknfnqgR7TJEzzeIVEf3aim8/view?usp=sharing

Thank you!

**Comment by Progi1984** [comment]
Thanks @matks & @khouloudbelguith

**Pull request merged**